kurtosis-module/impl/participant_network/el/mining_waiter/mining_waiter.go

```diff
@@ -20,6 +21,11 @@ func (waiter *miningWaiter) WaitForMining(numRetries uint32, timeBetweenRetries
 		if err == nil && blockNumber > 0 {
 			return nil
 		}
+		if err != nil {
```

Comment:
Just wondering, why not make this an `else if` off the prior conditional?

Reply:
chatted over Slack; has to do with the "eject early" in this case is actually ejecting on the _success_ case